### ISSUE TITLE
[gatsby-source-contentful] Passing arguments to a child node.

### ISSUE THREAD

**Issue by fpoumian** [gatsbyjs/gatsby#2984]
I have a Contentful model with the following Content Types:  Post, Tag and Category.

I want to query a specific Tag and also get the list of Posts that belong to that Tag, so I'm using the following GraphQL query:

```
query TagArchiveQuery($id: String!) {
    contentfulTag(id: { eq: $id }) {
      title
      slug
      post {
        title{
          title
        }
        tag {
          title
          slug
        }
      }
    }
  }
```

This query works as expected, however, ideally I would also like to be able to sort the list of Posts that are returned by that query (i.e. sort them by date).  I tried using the following query:

```
query TagArchiveQuery($id: String!) {
    contentfulTag(id: { eq: $id }) {
      title
      slug
      post(sort: { fields: [createdAt], order: DESC }) {
        title{
          title
        }
        tag {
          title
          slug
        }
      }
    }
  }
```
but I got the following error:

```
{
  "errors": [
    {
      "message": "Unknown argument \"sort\" on field \"post\" of type \"ContentfulTag\".",
      "locations": [
        {
          "line": 5,
          "column": 12
        }
      ]
    }
  ]
}
```

I'm relatively new to GraphQL so I'm not sure if I'm doing something wrong or it's just that the plugin does not support this type of queries.

**Comment by ryanwiemer** [comment]
@fpoumian I had a similar issue while I was getting familiar with GraphQL and working with Contentful. I too thought it would just be a matter of setting up a tag or category content type and then using a reference field in the post to associate it to a given tag / category. While that may work if you actually want the control of sorting or filtering via another filed I believe you have to set it up a little differently. Tag or category would be just be just a regular field rather than a separate content type. 

Here is an example query from a site I am working on now which pulls Galleries filtered by a category and then orders them by the date field:

```
query GalleryQuery {
    allContentfulGallery(limit: 1000, filter:{category:{eq:"lifestyle"}}, sort: { fields: [date], order: DESC }) {
    totalCount  
    edges {
        node {
          title
          slug
          date
          category
        }
      }
    }
  }
```

See this post on Stack Overflow for more info on this specific topic: https://stackoverflow.com/questions/47064987/gatsby-and-graphql-how-to-filter-array-within-query

I am still very new to this as well but hopefully this pushes you in the right direction.

**Comment by fpoumian** [comment]
@ryanwiemer Thanks for the tip, it's definitely helpful.

**Comment by calcsam** [comment]
Thanks @fpoumian -- we've been hearing more and more about folks new to GraphQL stumbling over this sort of error, appreciate you highlighting this. We're trying to think of how to make it easier for people, eg perhaps some sort of Gatsby-specific GraphQL query builder. Suggestions welcome!

**Comment by hunterbecton** [comment]
I came across this same issue and found that this Gatsby/Contentful Starter had a great example on how to order GraphQL data with lodash and moment: https://github.com/ryanwiemer/gatsby-starter-gcn/blob/master/src/templates/tag.js. You'll see that they created a new constant called `posts`

I also posted a more detailed answer in the stackoverflow link that @ryanwiemer mentioned above. I hope someone finds it useful!
